### ISSUE TITLE
Transpose is not working on version 0.20.2

### ISSUE THREAD

**Issue by MMin23** [SciSharp/NumSharp#348]
Hi,

I tested the transpose on version 0.106, it was working fine. However my transpose code is messed up when I switched to version 0.20.2. 

I am using this nunit test.

```
 public void Transpose3x2()
{
            NDArray x = np.arange(6).reshape(3, 2).MakeGeneric<int>();
            var y = np.transpose(x).MakeGeneric<int>();


            Assert.AreEqual(y[0, 0], 0);
            Assert.AreEqual(y[0, 1], 2);
            Assert.AreEqual(y[0, 2], 4);
            Assert.AreEqual(y[1, 0], 1);
            Assert.AreEqual(y[1, 1], 3);
            Assert.AreEqual(y[1, 2], 5);
}
```

**Comment by Nucs** [comment]
Duplicate of #117 

Hey @MMin23, currently we do not have transpose implemented, 0.106 is an old implementation that supports up to 3 dimensions I think.
I've finished a prototype of `np.swapaxes` (which is the base method for `np.transpose`) and probably push a new version to nuget tommorow with a full n-d implementation.

**Comment by Nucs** [comment]
`np.swapaxes`, `np.transpose`, `np.moveaxis` and `np.rollaxis` were added in the following commit: https://github.com/SciSharp/NumSharp/commit/39896599ac40584821802cfd63a15458060289ba 
And will be released soon in 0.20.4.
Thanks for holding up, it was quite challenging.
